packages/shared/src/components/sidebar/ContributeSection.tsx

```diff
@@ -46,39 +31,38 @@ export function ContributeSection({
       ),
       title: 'Submit article',
       action: trackAndShowSubmitArticle,
-      active: showSubmitArticle,
+      active: modal.type === LazyModal.SubmitArticle,
     },
     {
       icon: (active) => (
         <ListIcon Icon={() => <EmbedIcon secondary={active} />} />
       ),
       title: 'Suggest new source',
-      action: () => setShowNewSourceModal(true),
-      active: showNewSourceModal,
+      action: () => openModal({ type: LazyModal.NewSource }),
+      active: modal.type === LazyModal.NewSource,
     },
   ];
 
+  useEffect(() => {
+    const search = new URLSearchParams(window.location.search);
+    const query = Object.fromEntries(search);
+    if (!query?.scout) {
+      return;
+    }
+
+    const { origin, pathname } = window.location;
+    trackAndShowSubmitArticle();
```

Comment:
Yeah I think we can track this inside the modal rather?

Reply:
Relevant comment: https://github.com/dailydotdev/apps/pull/2160#discussion_r1321037239

Reply:
From inside the modal it would be under a `useEffect`. From here it would be right when the callback was executed. I think it would be better to avoid additional use effect. Wdyt?

Reply:
But this is a useEffect?
And what about other states that open the modal?
Are they non tracked

Reply:
This is a different use-case. This was coming from the deleted hook. This happens when the user received a notification that they are now eligible. They get redirected and we watch the query params to see if we should display.

Previously, we were actually not tracking it.